### PR TITLE
update dependencies and cleanup gradle files

### PR DESCRIPTION
**Pull request by FabianTerhorst**
update dependencies
replace jetpack dependence with bintray upload
remove mavenCentral
use http://semver.org/ for version numbers

### CODE REVIEW

**Comment by AbandonedCart** [comment]
Is semver just a GUI interface that tells you the version number or is the implementation of doing that going to be a separate PR?

**Comment by FabianTerhorst** [comment]
okhttp 3.4.1 is more stable to use so i updated it to 3.4.1. Its also the same version the google auth lib is using. For lombok the plugin will notify when the version is outdated.

File: library/build.gradle [hunk 1]
```diff
@@ -37,6 +24,8 @@ sourceSets {
     }
 }
 
+archivesBaseName = 'PokeGOAPI-library'
```
Comment (AbandonedCart) [inline library/build.gradle:29]: One way to allow more control from the master build script would be `archivesBaseName = archivesBaseName + '-library'` where the master build script has `archivesBaseName = 'PokeGOAPI'`

**Comment by FabianTerhorst** [comment]
#459

**Comment by AbandonedCart** [comment]
You noted `use http://semver.org/ for version numbers` but all I saw at the site was a thesis on how to choose version numbers, so is there more to it or was it just a reference?

**Comment by FabianTerhorst** [comment]
vi is my friend for squashes^^

**Comment by FabianTerhorst** [comment]
Its a reference for the scheme that should be used now

**Comment by FabianTerhorst** [comment]
every bugfix +1 on the last number and so on.

**Comment by FabianTerhorst** [comment]
every bugfix release*

**Comment by AbandonedCart** [comment]
I was aware how it works, just wasn't sure why it was a commit note. I thought it might be a service to handle incrementation. That's SOP.

**Pull request closed without merging**